src/table.js

```diff
@@ -659,10 +744,120 @@ export function __table(source, operations) {
       Object.fromEntries(operations.select.columns.map((c) => [c, d[c]]))
     );
   }
-  if (primitive) source = source.map((d) => d.value);
   if (source !== input) {
     if (schema) source.schema = schema;
     if (columns) source.columns = columns;
   }
   return source;
 }
+
+function coerceRow(object, types, schema) {
+  const coerced = {};
+  for (const col of schema) {
+    const type = types.get(col.name);
+    const value = object[col.name];
+    coerced[col.name] = type === "raw" ? value : coerceToType(value, type);
+  }
+  return coerced;
+}
+
+// https://github.com/d3/d3-dsv/issues/45
+const fixTz =
+  new Date("2019-01-01T00:00").getHours() ||
+  new Date("2019-07-01T00:00").getHours();
```

Comment:
This bug has been fixed in Safari, so I think we could safely omit this workaround to simplify the code here.

Reply:
@mbostock: Thanks! Have made these updates.